core-management/core-management-subsystem/src/main/java/org/wildfly/extension/core/management/UnstableApiAnnotationResourceDefinition.java

```diff
@@ -0,0 +1,103 @@
+import org.jboss.as.controller.descriptions.ModelDescriptionConstants;
+import org.jboss.as.controller.operations.validation.EnumValidator;
+import org.jboss.as.controller.registry.Resource;
+import org.jboss.as.server.AbstractDeploymentChainStep;
+import org.jboss.as.server.DeploymentProcessorTarget;
+import org.jboss.as.version.Stability;
+import org.jboss.dmr.ModelNode;
+import org.jboss.dmr.ModelType;
+import org.wildfly.extension.core.management.deployment.ReportUnstableApiAnnotationsProcessor;
+import org.wildfly.extension.core.management.deployment.ScanUnstableApiAnnotationsProcessor;
+
+/**
+ * Resource to configure the unstable api annotation usage reporting.
+ *
+ */
+public class UnstableApiAnnotationResourceDefinition extends PersistentResourceDefinition {
+
+    public static final Stability STABILITY = Stability.PREVIEW;
+    public static final SimpleAttributeDefinition LEVEL = SimpleAttributeDefinitionBuilder.create(
+            ModelDescriptionConstants.LEVEL, ModelType.STRING, true)
+            .setValidator(EnumValidator.create(UnstableApiAnnotationLevel.class))
+            .setDefaultValue(new ModelNode(UnstableApiAnnotationLevel.LOG.name()))
+            .setRestartAllServices()
+            .build();
+    public static final PathElement PATH = PathElement.pathElement(SERVICE, UNSTABLE_API_ANNOTATIONS);
+    static final ResourceRegistration RESOURCE_REGISTRATION = ResourceRegistration.of(PATH, STABILITY);
+    static final UnstableApiAnnotationResourceDefinition INSTANCE = new UnstableApiAnnotationResourceDefinition();
+
+    private static final List<AttributeDefinition> ATTRIBUTES = Collections.singletonList(LEVEL);
+
+    private UnstableApiAnnotationResourceDefinition() {
+        super(
+                new Parameters(
+                            RESOURCE_REGISTRATION,
+                            CoreManagementExtension.getResourceDescriptionResolver(UNSTABLE_API_ANNOTATIONS))
+                        .setAddHandler(UnstableApiAnnotationResourceAddHandler.INSTANCE)
+                        .setRemoveHandler(ReloadRequiredRemoveStepHandler.INSTANCE));
+    }
+
+
+    @Override
+    public Collection<AttributeDefinition> getAttributes() {
+        return ATTRIBUTES;
+    }
+
+    private static class UnstableApiAnnotationResourceAddHandler extends AbstractBoottimeAddStepHandler {
+
+
+        static final UnstableApiAnnotationResourceAddHandler INSTANCE = new UnstableApiAnnotationResourceAddHandler();
+        @Override
+        public void performBoottime(OperationContext context, ModelNode operation, Resource resource) throws OperationFailedException {
+            ModelNode model = resource.getModel();
+            String levelValue = UnstableApiAnnotationResourceDefinition.LEVEL.resolveModelAttribute(context, model).asString();
+            UnstableApiAnnotationLevel level = UnstableApiAnnotationLevel.valueOf(levelValue);
+
+            if (context.isNormalServer() && context.getStability().enables(STABILITY)) {
```

Comment:
FWIW, this stability check is redundant, since this code can only be reached if the stability level of the server enabled the registration of this resource and its operations.